File: modules/qm-foreground/src/QmForegroundModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class QmForegroundModule extends NativeModule {
  /** Start the foreground service with a plain notification. */
  start(title: string, body: string): void;
  /** Stop the foreground service + dismiss the notification. */
  stop(): void;
}

export default requireNativeModule<QmForegroundModule>('QmForeground');

File: modules/qm-foreground/android/src/main/java/expo/modules/qmforeground/QmForegroundModule.kt
package expo.modules.qmforeground

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class QmForegroundModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("QmForeground")

    Function("start") { title: String, body: String ->
      val ctx = appContext.reactContext
      if (ctx != null) QmForegroundService.start(ctx, title, body)
    }

    Function("stop") {
      val ctx = appContext.reactContext
      if (ctx != null) QmForegroundService.stop(ctx)
    }
  }
}
